File: h2o-docs/src/booklets/v2_2015/source/GLM_Vignette_code_examples/glm_compare_cross_validation_folds.py
h2o_df = h2o.import_file("http://s3.amazonaws.com/h2o-public-test-data/smalldata/airlines/allyears2k_headers.zip")
model = H2OGeneralizedLinearEstimator(family = "binomial", nfolds = 5)
model.train(y = "IsDepDelayed", x = ["Year", "Origin"], training_frame = h2o_df)


print "full model training auc:", model.auc()
print "full model cv auc:", model.auc(xval=True)
for model_ in model.get_xval_models():
    print model_.model_id, " training auc:", model_.auc(), " validation auc: ", model_.auc(valid=True)

File: h2o-docs/src/booklets/v2_2015/source/GLM_Vignette_code_examples/glm_compare_cross_validation_folds.R
library(h2o)
h2o.init()
h2o_df = h2o.importFile("http://s3.amazonaws.com/h2o-public-test-data/smalldata/airlines/allyears2k_headers.zip")
model = h2o.glm(y = "IsDepDelayed", x = c("Year", "Origin"), training_frame = h2o_df, family = "binomial", nfolds = 5, keep_cross_validation_models = TRUE)
print(paste("full model training auc:", model@model$training_metrics@metrics$AUC))
print(paste("full model cv auc:", model@model$cross_validation_metrics@metrics$AUC))
for (i in 1:5) {
    cv_model_name = model@model$cross_validation_models[[i]]$name
    cv_model = h2o.getModel(cv_model_name)
    print(paste("cv fold ", i, " training auc:", cv_model@model$training_metrics@metrics$AUC, " validation auc: ", cv_model@model$validation_metrics@metrics$AUC))
}
